sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuMultiFileReader.scala

```diff
@@ -1134,7 +1139,10 @@ abstract class MultiFileCoalescingPartitionReaderBase(
       logDebug(s"$getFileFormatShortName Coalescing reading estimates the initTotalSize:" +
         s" $initTotalSize, and the true size: $finalBufferSize")
       withResource(finalBuffer) { _ =>
-        finalBuffer.slice(0, finalBufferSize)
+        SpillableHostBuffer(
```

Comment:
Not directly related to this PR, but it reminds me that generally speaking, SpillableHostBuffer on a sliced HostMemoryBuffer could be, IMHO, buggy in some cases @abellina  @revans2 . Let's say originally I have a HostMemoryBuffer x, then another y sliced from x. When y is spilled and read from disk again, y is no longer sharing underlying buffer with x anymore, right? If the code writer assumes any content written to x is visible to y,  then SpillingFramework is causing a bug? Should we make sliced HostMemoryBuffer non-spillable at all?

Reply:
But marking sliced HostMemoryBuffer non-spillable at all has side effects, too. This will essentially make the original HostMemoryBuffer x also unspillable, because it always have another reference coming from the sliced HostMemoryBuffer

Reply:
Also,  SpillableHostBuffer on a sliced HostMemoryBuffer would mistakenly increase totalSize of SpillableHostStore: https://github.com/NVIDIA/spark-rapids/blob/b1e0fd696206147786daaef0401c3a1e02efddb2/sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala#L1363

Reply:
In this case,  we are slicing only to "right size" the buffer. The alternative is to memcpy it out to another right-sized buffer or to create a spillable buffer that knows about the slice, or track the "right size" alongside it for later slicing. 

In terms of tracking the memory, we'll be undercounting it because we have 0->finalBufferSize, not the whole length. We will let it go and spill it because we hold onto the only reference in the spill framework. When we bring it back after spill it will be of size `finalBufferSize`, not the original, but I think this is OK because we didn't mean to keep the original.

So I think the main issue is undercounting.

Reply:
Thx for the discussion, so should I make any update to fix this "undercounting"?

Reply:
@abellina For this case I think it's okay to keep the code as it is, because we can assume finalBufferSize is close to totalBufferSize. But I brought up this question to discuss the general cases ?

Reply:
OK, I fully get what is going on here. i will resolve this by "tracking the "right size" alongside it for later slicing".

Reply:
Updated